// spectacle/example/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _src = require("../src");

var _preloader = require("../src/utils/preloader");

var _preloader2 = _interopRequireDefault(_preloader);

var _default = require("../src/themes/default");

var _default2 = _interopRequireDefault(_default);

var _interactive = require("../assets/interactive");

var _interactive2 = _interopRequireDefault(_interactive);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

require("normalize.css");
require("../src/themes/default/index.css");

var images = {
  city: require("../assets/city.jpg"),
  kat: require("../assets/kat.png"),
  logo: require("../assets/formidable-logo.svg"),
  markdown: require("../assets/markdown.png")
};

(0, _preloader2.default)(images);

var theme = (0, _default2.default)({
  primary: "#ff4081"
});

var Presentation = function (_React$Component) {
  _inherits(Presentation, _React$Component);

  function Presentation() {
    _classCallCheck(this, Presentation);

    return _possibleConstructorReturn(this, (Presentation.__proto__ || Object.getPrototypeOf(Presentation)).apply(this, arguments));
  }

  _createClass(Presentation, [{
    key: "render",
    value: function render() {
      return _react2.default.createElement(
        _src.Spectacle,
        { theme: theme },
        _react2.default.createElement(
          _src.Deck,
          { transition: ["zoom", "slide"], transitionDuration: 500 },
          _react2.default.createElement(
            _src.Slide,
            { transition: ["zoom"], bgColor: "primary" },
            _react2.default.createElement(
              _src.Heading,
              { size: 1, fit: true, caps: true, lineHeight: 1, textColor: "black" },
              "Spectacle"
            ),
            _react2.default.createElement(
              _src.Heading,
              { size: 1, fit: true, caps: true },
              "A ReactJS Presentation Library"
            ),
            _react2.default.createElement(
              _src.Heading,
              { size: 1, fit: true, caps: true, textColor: "black" },
              "Where You Can Write Your Decks In JSX"
            ),
            _react2.default.createElement(
              _src.Link,
              { href: "https://github.com/FormidableLabs/spectacle" },
              _react2.default.createElement(
                _src.Text,
                { bold: true, caps: true, textColor: "tertiary" },
                "View on Github"
              )
            ),
            _react2.default.createElement(
              _src.Text,
              { textSize: "1.5em", margin: "20px 0px 0px", bold: true },
              "Hit Your Right Arrow To Begin!"
            )
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["slide"], bgColor: "black", notes: "You can even put notes on your slide. How awesome is that?" },
            _react2.default.createElement(_src.Image, { src: images.kat.replace("/", ""), margin: "0px auto 40px", height: "293px" }),
            _react2.default.createElement(
              _src.Heading,
              { size: 2, caps: true, fit: true, textColor: "primary", textFont: "primary" },
              "Wait what?"
            )
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["zoom", "fade"], bgColor: "primary", notes: "<ul><li>talk about that</li><li>and that</li></ul>" },
            _react2.default.createElement(_src.CodePane, {
              lang: "jsx",
              source: require("raw!../assets/deck.example"),
              margin: "20px auto"
            })
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["slide"], bgImage: images.city.replace("/", ""), bgDarken: 0.75 },
            _react2.default.createElement(
              _src.Appear,
              { fid: "1" },
              _react2.default.createElement(
                _src.Heading,
                { size: 1, caps: true, fit: true, textColor: "primary" },
                "Full Width"
              )
            ),
            _react2.default.createElement(
              _src.Appear,
              { fid: "2" },
              _react2.default.createElement(
                _src.Heading,
                { size: 1, caps: true, fit: true, textColor: "tertiary" },
                "Adjustable Darkness"
              )
            ),
            _react2.default.createElement(
              _src.Appear,
              { fid: "3" },
              _react2.default.createElement(
                _src.Heading,
                { size: 1, caps: true, fit: true, textColor: "primary" },
                "Background Imagery"
              )
            )
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["zoom", "fade"], bgColor: "primary" },
            _react2.default.createElement(
              _src.Heading,
              { caps: true, fit: true },
              "Flexible Layouts"
            ),
            _react2.default.createElement(
              _src.Layout,
              null,
              _react2.default.createElement(
                _src.Fill,
                null,
                _react2.default.createElement(
                  _src.Heading,
                  { size: 4, caps: true, textColor: "secondary", bgColor: "white", margin: 10 },
                  "Left"
                )
              ),
              _react2.default.createElement(
                _src.Fill,
                null,
                _react2.default.createElement(
                  _src.Heading,
                  { size: 4, caps: true, textColor: "secondary", bgColor: "white", margin: 10 },
                  "Right"
                )
              )
            )
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["slide"], bgColor: "black" },
            _react2.default.createElement(
              _src.BlockQuote,
              null,
              _react2.default.createElement(
                _src.Quote,
                null,
                "Wonderfully formatted quotes"
              ),
              _react2.default.createElement(
                _src.Cite,
                null,
                "Ken Wheeler"
              )
            )
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["spin", "zoom"], bgColor: "tertiary" },
            _react2.default.createElement(
              _src.Heading,
              { caps: true, fit: true, size: 1, textColor: "primary" },
              "Inline Markdown"
            ),
            _react2.default.createElement(
              _src.Markdown,
              null,
              "\n![Markdown Logo](" + images.markdown.replace("/", "") + ")\n\nYou can write inline images, [Markdown Links](http://commonmark.org), paragraph text and most other markdown syntax\n* Lists too!\n* With ~~strikethrough~~ and _italic_\n* And let's not forget **bold**\n              "
            )
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["slide", "spin"], bgColor: "primary" },
            _react2.default.createElement(
              _src.Heading,
              { caps: true, fit: true, size: 1, textColor: "tertiary" },
              "Smooth"
            ),
            _react2.default.createElement(
              _src.Heading,
              { caps: true, fit: true, size: 1, textColor: "secondary" },
              "Combinable Transitions"
            )
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["fade"], bgColor: "secondary", textColor: "primary" },
            _react2.default.createElement(
              _src.List,
              null,
              _react2.default.createElement(
                _src.Appear,
                null,
                _react2.default.createElement(
                  _src.ListItem,
                  null,
                  "Inline style based theme system"
                )
              ),
              _react2.default.createElement(
                _src.Appear,
                null,
                _react2.default.createElement(
                  _src.ListItem,
                  null,
                  "Autofit text"
                )
              ),
              _react2.default.createElement(
                _src.Appear,
                null,
                _react2.default.createElement(
                  _src.ListItem,
                  null,
                  "Flexbox layout system"
                )
              ),
              _react2.default.createElement(
                _src.Appear,
                null,
                _react2.default.createElement(
                  _src.ListItem,
                  null,
                  "PDF export"
                )
              ),
              _react2.default.createElement(
                _src.Appear,
                null,
                _react2.default.createElement(
                  _src.ListItem,
                  null,
                  "And..."
                )
              )
            )
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["slide"], bgColor: "primary" },
            _react2.default.createElement(
              _src.Heading,
              { size: 1, caps: true, fit: true, textColor: "tertiary" },
              "Your presentations are interactive"
            ),
            _react2.default.createElement(_interactive2.default, null)
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["slide"], bgColor: "primary",
              notes: "Hard to find cities without any pizza"
            },
            _react2.default.createElement(
              _src.Heading,
              { size: 4, caps: true, textColor: "secondary", bgColor: "white", margin: 10 },
              "Pizza Toppings"
            ),
            _react2.default.createElement(
              _src.Layout,
              null,
              _react2.default.createElement(
                _src.Table,
                null,
                _react2.default.createElement(
                  "thead",
                  null,
                  _react2.default.createElement(
                    _src.TableRow,
                    null,
                    _react2.default.createElement(_src.TableHeaderItem, null),
                    _react2.default.createElement(
                      _src.TableHeaderItem,
                      null,
                      "2011"
                    ),
                    _react2.default.createElement(
                      _src.TableHeaderItem,
                      null,
                      "2013"
                    ),
                    _react2.default.createElement(
                      _src.TableHeaderItem,
                      null,
                      "2015"
                    )
                  )
                ),
                _react2.default.createElement(
                  "tbody",
                  null,
                  _react2.default.createElement(
                    _src.TableRow,
                    null,
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "None"
                    ),
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "61.8%"
                    ),
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "39.6%"
                    ),
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "35.0%"
                    )
                  ),
                  _react2.default.createElement(
                    _src.TableRow,
                    null,
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "Pineapple"
                    ),
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "28.3%"
                    ),
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "54.5%"
                    ),
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "61.5%"
                    )
                  ),
                  _react2.default.createElement(
                    _src.TableRow,
                    null,
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "Pepperoni"
                    ),
                    _react2.default.createElement(_src.TableItem, null),
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "50.2%"
                    ),
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "77.2%"
                    )
                  ),
                  _react2.default.createElement(
                    _src.TableRow,
                    null,
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "Olives"
                    ),
                    _react2.default.createElement(_src.TableItem, null),
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "24.9%"
                    ),
                    _react2.default.createElement(
                      _src.TableItem,
                      null,
                      "55.9%"
                    )
                  )
                )
              )
            )
          ),
          _react2.default.createElement(
            _src.Slide,
            { transition: ["spin", "slide"], bgColor: "tertiary" },
            _react2.default.createElement(
              _src.Heading,
              { size: 1, caps: true, fit: true, lineHeight: 1.5, textColor: "primary" },
              "Made with love in Seattle by"
            ),
            _react2.default.createElement(
              _src.Link,
              { href: "http://www.formidable.com" },
              _react2.default.createElement(_src.Image, { width: "100%", src: images.logo })
            )
          )
        )
      );
    }
  }]);

  return Presentation;
}(_react2.default.Component);

exports.default = Presentation;
